packages/web/src/components/usdc-purchase-details-modal/USDCPurchaseDetailsModal.tsx

```diff
@@ -0,0 +1,247 @@
+}: {
+  children: React.ReactNode
+  label: string
+}) => (
+  <div className={styles.detailSection}>
+    <Text variant='label' size='large' color='neutralLight4'>
+      {label}
+    </Text>
+    {children}
+  </div>
+)
+
+const TrackLink = ({ id, onClick }: { id: number; onClick: () => void }) => {
+  const { data: track } = useGetTrackById({ id })
+  if (!track) return null
+  return (
+    <Link onClick={onClick} className={styles.link} to={track.permalink}>
+      <Text size='large' color='secondary'>
+        {track.title}
+      </Text>
+    </Link>
+  )
+}
+
+const PurchaseModalContent = ({ purchaseDetails, onClose }: ContentProps) => {
+  const goToRoute = useGoToRoute()
+  const { data: track } = useGetTrackById({ id: purchaseDetails.contentId })
+  const onClickVisitTrack = useCallback(() => {
+    if (track) {
+      onClose()
+      goToRoute(track.permalink)
+    }
+  }, [track, onClose, goToRoute])
+
+  return (
+    <>
+      <ModalHeader>
+        <ModalTitle
+          iconClassName={styles.icon}
+          icon={<Icon icon={IconCart} />}
+          title={messages.purchaseDetails}
+        />
+      </ModalHeader>
+      <ModalContent className={styles.content}>
+        <div className={styles.trackRow}>
+          <DetailSection label={messages.track}>
+            <TrackLink onClick={onClose} id={purchaseDetails.contentId} />
+          </DetailSection>
+          <DynamicTrackArtwork id={purchaseDetails.contentId} />
+        </div>
+        <DetailSection label={messages.by}>
+          <Text size='large' color='secondary'>
+            <UserNameAndBadges
+              onNavigateAway={onClose}
+              userId={purchaseDetails.sellerUserId}
+            />
+          </Text>
+        </DetailSection>
+        <DetailSection label={messages.date}>
+          <Text size='large'>
+            {moment(purchaseDetails.createdAt).format('M/D/YY')}
+          </Text>
+        </DetailSection>
+        <DetailSection label={messages.cost}>
+          <Text size='large'>{`$${formatUSDCWeiToUSDString(
+            purchaseDetails.amount
+          )}`}</Text>
+        </DetailSection>
+      </ModalContent>
+      <ModalFooter className={styles.footer}>
+        <HarmonyButton
+          className={styles.button}
+          variant={HarmonyButtonType.GHOST}
+          text={messages.visitTrack}
+          iconRight={IconArrow}
+          onClick={onClickVisitTrack}
+        />
+        <HarmonyButton
+          className={styles.button}
+          text={messages.done}
+          onClick={onClose}
+        />
+      </ModalFooter>
+    </>
+  )
+}
+
+const SaleModalContent = ({ purchaseDetails, onClose }: ContentProps) => {
+  const dispatch = useDispatch()
+  const { onOpen: openInboxUnavailableModal } = useInboxUnavailableModal()
+
+  const { canCreateChat } = useSelector((state: CommonState) =>
+    getCanCreateChat(state, { userId: purchaseDetails.buyerUserId })
+  )
+
+  const onClickMessageBuyer = useCallback(() => {
```

Comment:
really nice to see all these hooks coming together!

Reply:
`handleClickMessageBuyer` just nit